frontend/src/components/FormGroupSettings.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import {
+  FormGroup,
+  Text,
+  HelperText,
+  HelperTextItem,
+  Alert,
+  AlertActionCloseButton,
+  Hint,
+  HintBody,
+} from '@patternfly/react-core';
+import { MultiSelection } from './MultiSelection';
+import { GroupsConfigField, MenuItemStatus } from 'pages/groupSettings/groupTypes';
+
+type FormGroupSettingsProps = {
+  title: string;
+  body: string;
+  footer: string;
+  fieldId: string;
+  groupsField: GroupsConfigField;
+  items: MenuItemStatus[];
+  handleMenuItemSelection: (newState: MenuItemStatus[], field: GroupsConfigField) => void;
+  handleClose: () => void;
+  error?: string;
+  hintText?: string;
+};
+
+export const FormGroupSettings: React.FC<FormGroupSettingsProps> = ({
+  title,
+  body,
+  footer,
+  fieldId,
+  groupsField,
+  items,
+  handleMenuItemSelection,
+  handleClose,
+  error,
+  hintText,
+}) => {
+  return (
+    <FormGroup fieldId={fieldId} label={title}>
+      <Text>{body}</Text>
+      <MultiSelection
+        value={items}
+        setValue={(newState) => handleMenuItemSelection(newState, groupsField)}
+        clearSelection={() => {
+          const newState = items?.map((element) => ({ ...element, enabled: false }));
+          handleMenuItemSelection(newState, groupsField);
+        }}
+      />
+      {!error && (
+        <>
+          <HelperText>
+            <HelperTextItem variant="indeterminate">{footer}</HelperTextItem>
+          </HelperText>
+          {hintText && (
+            <Hint>
+              <HintBody>{hintText}</HintBody>
+            </Hint>
+          )}
+        </>
+      )}
+      {error && (
+        <Alert
+          isInline
+          variant="warning"
+          title={`Group no longer exists`}
```

Comment:
Is this the only type of error we can have? I would have expected a more generic title 🤔